### PR TITLE
replace create-react-app with create-react-library

### PR DESCRIPTION
**Pull request by orlando**
**What:** replace create-react-app with create-react-library

**Why:** work related to https://github.com/debtcollective/disputes/issues/14

**How:** created a project with **create-react-library** and moved the files over the header repo once everything was working

### create-react-library
[create-react-library](https://github.com/transitive-bullshit/create-react-library) is better oriented to create React component libraries than create-react-app. It is based on Rollup which now seems to be the default for creating components (vs webpack which is more oriented to applications).

The build process now works correctly, managing `peerDependencies` correctly, you can see it working in the `example` folder.

### pretty-quick

**pretty-quick** doesn't have integration with ESLint (https://github.com/azz/pretty-quick/issues/22). The problem was that pretty-quick had a different output than the eslint command, and it wasn't using the eslint configuration, which is not what we want. 
Prettier has [prettier-eslint](https://github.com/prettier/prettier-eslint), we are using through [prettier-eslint-cli](https://github.com/prettier/prettier-eslint-cli) instead of pretty-quick. This runs prettier and ESLint in one pass. 

Let's use this setup in other projects and tweak it as we need it.

### gh-pages

We are deploying the example to Github pages https://debtcollective.github.io/header/. This should be done by the CI every time we merge to master.

### CODE REVIEW

File: src/.eslintrc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (duranmla) [inline src/.eslintrc]: Why do we have two files of `.eslintrc`?

File: src/common/DiscourseService.js [hunk 2]
```diff
@@ -1,13 +1,8 @@
...
+// TODO: We should find a better way to do this, ex: passing this value as a prop.
 const setDiscourseEndpoint = () => {
-  const discourseEndpoint = process.env.REACT_APP_DISCOURSE_ENDPOINT;
-
-  if (!discourseEndpoint) {
-    throw new Error("You must provide a valid Discourse enpoint");
-  }
-
-  return discourseEndpoint;
+  return process.env.DISCOURSE_ENDPOINT || "http://localhost:3000";
```
Comment (duranmla) [inline src/common/DiscourseService.js]: So we built passing an environment variable I guess right?

**Comment by duranmla** [comment]
This looks great, couple questions but nothing blocker.
I do not know if it is worthy to add some pending tasks to revise later related to:
- Check the usage of the environment variable to see if worthy to use config based approach
- `docz` currently we do not have a way to make sure the docz are not broken when we push to master (a sort of checking if it built correctly or publishing - I do not know if worthy neither -)

**Comment by duranmla** [comment]
Nice work 👌

File: package.json [hunk 3]
```diff
@@ -1,93 +1,122 @@
...
-    "emotion-theming": "10.0.6",
-    "eslint-plugin-flowtype": "3.7.0",
-    "eslint-plugin-react": "7.13.0",
+    "eslint": "^5.0.1",
+    "eslint-config-standard": "^11.0.0",
+    "eslint-config-standard-react": "^6.0.0",
+    "eslint-plugin-flowtype": "~3.8.1",
+    "eslint-plugin-import": "^2.13.0",
+    "eslint-plugin-node": "^7.0.1",
+    "eslint-plugin-promise": "^4.0.0",
```
Comment (duranmla) [inline package.json]: BTW, having these plugins from eslint but not in the configuration sounds like we are not actually using it.
Are you positive about their usage? I just see in the config flow and react

File: src/.eslintrc (hunk 1 above)
Comment (orlando) [reply src/.eslintrc]: Good question, this was there when I copied the files over. I’ll remove it

File: package.json (hunk 3 above)
Comment (orlando) [reply package.json]: Let’s talk about this, by default this usess standard js and I would like to use something like that

File: src/common/DiscourseService.js (hunk 2 above)
Comment (orlando) [reply src/common/DiscourseService.js]: Yeah, we need to check how to make this work. If there's more work to do when integrating with the disputes app I'll create a ticket for it

**Pull request merged**